backend/dal/Services/Concrete/UserService.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using Pims.Dal.Entities;
+using System.Security.Claims;
+using Pims.Dal.Helpers.Extensions;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+
+namespace Pims.Dal.Services.Concrete
+{
+    //TODO: Comments
+    public class UserService : IUserService
+    {
+        private readonly PIMSContext _dbContext;
+        private readonly ClaimsPrincipal _user;
+        public UserService(PIMSContext dbContext, ClaimsPrincipal user)
+        {
+            _dbContext = dbContext;
+            _user = user;
+        }
+
+        public IEnumerable<User> GetUsers()
+        {
+            var query = _dbContext.Users.AsNoTracking();
+            return query.ToArray();
+        }
+
+        public User GetUser(Guid id)
+        {
+            return _dbContext.Users.AsNoTracking().FirstOrDefault(u => u.Id == id);
+        }
+
+        public User AddUser(User entity)
+        {
+            var userId = this._user.GetUserId();
+            entity.CreatedById = userId;
+            _dbContext.Users.Add(entity);
+            _dbContext.CommitTransaction();
+            return entity;
+        }
+
+        public User UpdateUser(Guid Id, User entity)
+        {
+            var entityToUpdate = _dbContext.Users.Find(Id);
+
+            if (entity == null) return null;
+            var userId = this._user.GetUserId();
+
+            entity.UpdatedOn = DateTime.UtcNow;
+            entity.UpdatedById = userId;
+            _dbContext.Entry(entityToUpdate).CurrentValues.SetValues(entity);
+            _dbContext.Users.Update(entity);
+            _dbContext.CommitTransaction();
+            return entityToUpdate;
+        }
+
+        public User DeleteUser(Guid id, User user)
```

Comment:
You shouldn't need to include the Id like this.  As the Id should be in the entity.  It's also impossible to change a Primary Key value in an update.  Change to `DeleteUser(User entity)`